doc/index.rst

```diff
@@ -0,0 +1,366 @@
+
+.. code-block:: py
+   :name: gs-butler-get
+
+   from lsst.daf.persistence import Butler
+   butler = Butler('./output')
+   calexp = butler.get('calexp', run=4192, camcol=4, field=300, filter='g')
+
+The next lines will set up the display, using a public Firefly server at IPAC.
+
+.. code-block:: py
+   :name: gs-setup-display
+
+   import lsst.afw.display as afw_display
+   display1 = afw_display.Display(frame=1, backend='firefly',
+      host='irsa.ipac.caltech.edu', port=80, basedir='irsaviewer',
+      channel='mychannel')
+
+Open a browser window to
+`http://irsa.ipac.caltech.edu/irsaviewer/;wsch=mychannel <http://irsa.ipac.caltech.edu/irsaviewer/;wsch=mychannel>`_.
+Then display the exposure:
+
+.. code-block:: py
+   :name: gs-display-calexp
+
+   display1.mtv(calexp)
+
+The displayed exposure will appear in your browser.
+
+.. _lsst-display-firefly-using:
+
+Using lsst.display.firefly
+==========================
+
+The `lsst.display.firefly` package is not used directly. Instead, instances
+of `lsst.afw.display` are created with parameters passed to the backend.
+The Firefly backend is based upon the Python API in
+:class:`firefly_client.FireflyClient`.
+
+Setup
+-----
+
+Before a Python session or Jupyter notebook is started, setup of stack
+packages must be completed. Use :command:`setup display_firefly` to enable
+the Firefly backend.
+
+Initializing with lsst.afw.display
+----------------------------------
+
+The recommended way to create a display object for Firefly is using
+the :class:`Display` constructor from `lsst.afw.display`:
+
+.. code-block:: py
+    :name: construct-display
+
+    import lsst.afw.display as afw_display
+    display1 = afw_display.Display(frame=1, host='localhost', port=8080,
+                                   basedir='firefly', channel='afw')
+
+The parameters shown above (besides ``frame``) are the defaults and will
+apply when running a Firefly server locally with default settings.
+
+If a Firefly server has been provided to you, set ``host``, ``port``, and
+``basedir`` according to the informatio provided. You should set ``channel``
+to a unique string to avoid another user from writing to your display.
+
+.. warning::
+
+   Once a :class:`Display` instance is made, within your Python session
+   it will not be possible to define another display pointing to a different
+   server.
+
+Opening a browser window
+------------------------
+
+A browser window or tab must be opened before any data are displayed.
+
+When using a Firefly server on ``localhost``, creating the display object
+will cause a browser window to open to the correct location. For this
+example, the ``display1.show()`` method is run.
+
+When using a server with a different host name, you will likely have to
+open a browser window or tab on your local machine yourself. For example,
+for ``host=lsst-dev``, ``port=8085``, ``basedir=firefly``, ``channel=mine``,
+use the url ``http://lsst-dev:8085/firefly/;wsch=mine``.
```

Comment:
It's surprising to me that if the constructor can open a window pointing to localhost that it can't open a window pointing someplace else.

Reply:
Currently the constructor tests for localhost. If the Python session is remote, there is no good way to cause a browser to open on your local machine. I have seen instances where `display1.show()` on a remote machine will start a browser remotely and have it display via XWindows...in that case you really do want the local browser instead.